File: frontend/src/services/analysis.ts
export interface AskQuestionResponse {
  answer: string
  sql: string
  row_count: number
}

export async function askQuestion(question: string): Promise<AskQuestionResponse> {
  const res = await fetch('/api/analysis/ask', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ question }),
  })
  if (!res.ok) {
    const detail = await res.json().catch(() => null)
    throw new Error(detail?.detail ?? `${res.status}`)
  }
  return res.json() as Promise<AskQuestionResponse>
}

File: backend/routers/analysis.py
from fastapi import APIRouter, Depends, HTTPException

from backend.database.models import User
from backend.schemas import AskQuestionRequest, AskQuestionResponse
from backend.services.ai_analysis import AiAnalysisError, ask_question
from backend.services.auth import get_current_user

router = APIRouter()


@router.post("/analysis/ask", response_model=AskQuestionResponse)
def ask(body: AskQuestionRequest, user: User = Depends(get_current_user)) -> AskQuestionResponse:
    try:
        result = ask_question(body.question)
    except AiAnalysisError as exc:
        raise HTTPException(status_code=422, detail=str(exc)) from exc
    return AskQuestionResponse(answer=result.answer, sql=result.sql, row_count=result.row_count)
